### DISCUSSION TITLE
Run locally directus development version error with dependency

### DISCUSSION THREAD

**Discussion by markgiul** [directus/directus#4932] [Archive]
Hi,
i'm trying to install locally directus i'm using the guide https://docs.directus.io/contributing/running-locally/ but when i launch this command `npm run cli bootstrap` i get an error `Error: Cannot find module '@directus/format-title'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:613:15)
    at Function.Module._load (internal/modules/cjs/loader.js:539:25)
    at Module.require (internal/modules/cjs/loader.js:667:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (Projects/directus-9-development/api/src/database/migrations/run.ts:4:1)
    at Module._compile (internal/modules/cjs/loader.js:738:30)
    at Module.m._compile (Projects/directus-9-development/node_modules/ts-node/src/index.ts:1056:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:749:10)
    at Object.require.extensions.(anonymous function) [as .ts] (Projects/directus-9-development/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (internal/modules/cjs/loader.js:630:32)
`, even if i try to launch the serve locally with `npm run dev` it gives me errors that always concerns the same dependency and also another dependency `
@directus/app:  INFO  Starting development server...
directus: (node:38227) UnhandledPromiseRejectionWarning: Error: Cannot find module '@directus/drive'
directus:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:613:15)
directus:     at Function.Module._load (internal/modules/cjs/loader.js:539:25)
directus:     at Module.require (internal/modules/cjs/loader.js:667:17)
directus:     at require (internal/modules/cjs/helpers.js:20:18)
directus:     at Object.<anonymous> (Projects/directus-9-development/api/src/storage.ts:1:1)
directus:     at Module._compile (internal/modules/cjs/loader.js:738:30)
directus:     at Module._compile (Projects/directus-9-development/node_modules/source-map-support/source-map-support.js:547:25)
directus:     at Module.m._compile (/private/var/folders/6p/1p8bdqmd1b78_zjt1fj1qzlw0000gp/T/ts-node-dev-hook-6548969625245904.js:69:33)
directus:     at Module._extensions..js (internal/modules/cjs/loader.js:749:10)
directus:     at require.extensions.(anonymous function) (/private/var/folders/6p/1p8bdqmd1b78_zjt1fj1qzlw0000gp/T/ts-node-dev-hook-6548969625245904.js:71:20)
directus: (node:38227) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
directus: (node:38227) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`, but this i think because i didn't init the database correctly.
I do not understand why these dependencies are not found, could you give me some suggestions.
Thanks

**Comment by nickrum** [comment]
Which npm version are you using? Make sure you are on npm 7+.

**Reply by markgiul** [reply]
Now using node v15.14.0 (npm v7.7.6)

**Reply by nickrum** [reply]
Alright, did the `npm install` call succeed?

**Reply by markgiul** [reply]
yes the install is correctly succeed.

**Reply by nickrum** [reply]
Could you try to run `npm run build` in the root directory of the monorepo?

**Reply by markgiul** [reply]
no but I'll do it now

**Reply by markgiul** [reply]
now after build, i run `npm run cli bootstrap` and worked correctly.I have also launched the dev server and it has started, now I am left to figure out how to access the application in vue, can you suggest how to do this. Perhaps it should be suggested in the guide, thanks a lot.

**Reply by nickrum** [reply]
Not sure exactly what you mean. If you want to access the admin app, you can use the URL logged to your terminal after calling `npm run dev`. If you want to access the source code of the admin app, it's in the `app/` directory.

**Reply by markgiul** [reply]
I didn't see the url in the terminal, I entered and everything was ok, however I noticed in the browser console that the app doesn't find the custom extensions, but if you know, in development mode is it possible to make the app access the call `localhost:8081/extensions/layouts|modules...` because now the call goes 404, so I was wondering if in the proxy of the app there is some configuration that doesn't allow access to this api, also I checked that the server returns the loaded extensions and they are returned but from the app they don't go, but I don't know if it's possible with the monorepo to let the app access to the folder of the files built in the api.
I say this because my ultimate goal is to develop custom extensions and in some of them I need to access the internal components of the app for example the client api or the collection configurations.

**Reply by nickrum** [reply]
In theory that should work, there might be a bug that currently prevents this from working correctly though (see #4902).
